First Pass at ONCALL Template

polynomial opened the pull request.

Commit:
First Pass at ONCALL Template

mbbx6spp commented on docs/ONCALL.adoc:
```diff
@@ -0,0 +1,13 @@
+
+=== Issues
+
+* Put each issue responded to here. There should be a link to the ticket
```
I would like to see the following metrics/breakdowns of the issues from the week:
- **Action summary breakdown:** this would categorize each issue into one of the following buckets:
  - no action taken, on-going maintenance (expired window) - link to CCR
  - no action taken, on-going maintenance (no window set) - link to CCR
  - no action taken, threshold adjustment needed - link to change of threshold in code or monitoring system
  - no action taken, check is not reliable/needs updating - link to ticket to fix check script
  - action taken - link to ticket with investigative comments/notes
- **Sleep state summary:** this needs to be specified by the on-call responder as to whether the alert came in while they were asleep or awake.

There are a couple more that we might add eventually but these will be a good start to understanding our on-call issues.

mbbx6spp commented on docs/ONCALL.adoc:
```diff
@@ -0,0 +1,13 @@
+
+=== Continuing Issues
+
+* Put issues that weren't fully resolved here, include a link and a risk summary
```
This could be derived (read: generated) from the classifications below and it would provide context for why they are continuing with necessary work links.

mbbx6spp commented on docs/ONCALL.adoc:
```diff
@@ -0,0 +1,13 @@
+
+=== Summary/Notes:
+
+* Use this section to put high level context that may be helpful to the next person on-call
```
It would be great to provide an illustrative example for this. For example:

```
Last week (Friday night) there was one product launch and two large releases for existing products/services (Tuesday night). Many of the incidents triggered were around these code changes in production. Some (~30%) were false positives due to outmoded status scripts for infrastructure.
```

mbbx6spp commented:
It would be prudent to use consistent terminology (and document meanings). I personally prefer PagerDuty's _incident_ (as opposed to issue). An issue might link multiple _incidents_ together because it is the same root cause. A ticket refers to a ticket in the ticketing system being used to track work around on-call activities. An _alert_ is the individual call, SMS, text message, email, etc used to notify the person(s) on-call.

mbbx6spp commented:
Also at some point we need PD scripts to automate the scraping of the last on-call period of incidents so the on-call person can categorize/classify the incidents as per the breakdowns above.

polynomial commented:
@mbbx6spp thank you for the comments. I agree about the scraping/generated stuff. For the other comments I made changes. For the ones about collecting breakdowns I changed how the (now named) Incidents are collected. I put some BNF in there as a rough way of shaping how I was thinking about it and added an example to that section as well (thanks for the suggestion!). Hopefully this going in the right direction, but its a bigger change.

mbbx6spp commented on docs/ONCALL.adoc:
```diff
@@ -0,0 +1,37 @@
+** Summary - Action Taken - Ticket Link [Incident Link] - Quality of Life Impact - Exceptions
+** Options for each Section:
+*** Summary: one sentence description of the problem
+*** Action: Action Taken (Further Investigation Needed | Other, Ticket Link ) | No Action Taken (on-going maintenance ( inside | outside ) window | False Positive, Needs Threshold Adjustment)
```
I still prefer my action categories. To me an action should be defined as something to resolve the _incident_ that the on-call responder was notified about. We are unable to differentiate between bad check scripts and bad/changing thresholds.

mbbx6spp commented on docs/ONCALL.adoc:
```diff
@@ -0,0 +1,37 @@
+** Options for each Section:
+*** Summary: one sentence description of the problem
+*** Action: Action Taken (Further Investigation Needed | Other, Ticket Link ) | No Action Taken (on-going maintenance ( inside | outside ) window | False Positive, Needs Threshold Adjustment)
+*** Quality of Life Impact: [Lost Sleep]
```
While this might seem minor the awake/asleep state of the on-call responder is a factual statement. "Lost sleep" is potentially subjective and open to interpretation by different people. For example, if someone received an alert for an incident at 1am and they were up anyway, it should count as awake, but some people may just think "it's in sleep hours, so I will put 'Lost Sleep'". We should use awake/asleep. :)

mbbx6spp commented on docs/ONCALL.adoc:
```diff
@@ -0,0 +1,37 @@
+*** Summary: one sentence description of the problem
+*** Action: Action Taken (Further Investigation Needed | Other, Ticket Link ) | No Action Taken (on-going maintenance ( inside | outside ) window | False Positive, Needs Threshold Adjustment)
+*** Quality of Life Impact: [Lost Sleep]
+*** Exceptions: Escalation Out of Band | Missing Documentation | Unclear Owner/Subject Matter Expert | Missing Diagnostics
```
I would prefer to call this 'Further Work' instead of 'Exceptions'. Links to tickets for this should be provided too.

mbbx6spp commented on docs/ONCALL.adoc:
```diff
@@ -0,0 +1,37 @@
+*** Exceptions: Escalation Out of Band | Missing Documentation | Unclear Owner/Subject Matter Expert | Missing Diagnostics
+
+Example:
+Riak Performance Problem - Action Taken Added More Metrics: OPS-1234 - None - Missing Diagnostics
```
Minor: use table formatting.
